src/java.base/share/classes/java/lang/invoke/DirectMethodHandle.java

```diff
@@ -896,11 +896,11 @@ private static NamedFunction createFunction(byte func) {
                 case NF_constructorMethod:
                     return getNamedFunction("constructorMethod", OBJ_OBJ_TYPE);
                 case NF_UNSAFE:
-                    MemberName member = new MemberName(MethodHandleStatics.class, "UNSAFE", Unsafe.class, REF_getField);
+                    MemberName member = new MemberName(MethodHandleStatics.class, "UNSAFE", Unsafe.class, REF_getStatic);
                     return new NamedFunction(
                             MemberName.getFactory().resolveOrFail(REF_getField, member,
```

Comment:
`REF_getField` passed to `resolveOrFail` should also be corrected?

Reply:
Yes, fixed.